cmd/main.go

```diff
@@ -35,31 +39,49 @@ func PipelineAction(pf pipelines.PipelineFunc) cli.ActionFunc {
 			ctx  = c.Context
 			opts = []dagger.ClientOpt{}
 		)
+		ctx, span := otel.Tracer("grafana-build").Start(ctx, fmt.Sprintf("pipeline-%s", c.Command.Name))
+		defer span.End()
+
 		if c.Bool("verbose") {
 			opts = append(opts, dagger.WithLogOutput(os.Stderr))
 		}
 		client, err := dagger.Connect(ctx, opts...)
 		if err != nil {
+			span.RecordError(err)
+			span.SetStatus(codes.Error, "failed to connect to Dagger")
 			return err
 		}
+		defer client.Close()
 
-		args, err := pipelines.PipelineArgsFromContext(c.Context, c)
+		args, err := pipelines.PipelineArgsFromContext(ctx, c)
 		if err != nil {
+			span.RecordError(err)
+			span.SetStatus(codes.Error, "failed to load arguments")
```

Comment:
The amount of times this is repeated makes me think we need a wrapper function, don't you think?

Reply:
Done :)